synapse/storage/_base.py

```diff
@@ -352,14 +360,14 @@ def set_expiration_date_for_user_txn(self, txn, user_id, use_delta=False):
         )
 
     def start_profiling(self):
-        self._previous_loop_ts = self._clock.time_msec()
+        self._previous_loop_ts = monotonic_time()
 
         def loop():
             curr = self._current_txn_total_time
             prev = self._previous_txn_total_time
             self._previous_txn_total_time = curr
 
-            time_now = self._clock.time_msec()
+            time_now = monotonic_time()
```

Comment:
so yes. this function turned out to be passing milliseconds into a function that expected seconds.